model/validation.go

```diff
@@ -204,16 +204,367 @@ func (t Tags) Validate() error {
+		err = multierror.Append(err, fmt.Errorf("%v must be in range 0..100", label))
+	}
+	return err
+}
+
+func validateFloatPercent(err error, val float32, label string) error {
+	if val > 100.0 {
+		err = multierror.Append(err, fmt.Errorf("%v must not exceed 100", label))
+	}
+	if val < 0.0 {
+		err = multierror.Append(err, fmt.Errorf("%v must be in range 0..100", label))
+	}
+	return err
+}
+
+// ValidateDestinationWeight validates DestinationWeight
+func ValidateDestinationWeight(dw *proxyconfig.DestinationWeight) error {
+	var retVal error
+
+	if dw.Destination != "" {
+		if err := validateFQDN(dw.Destination); err != nil {
+			retVal = multierror.Append(retVal, err)
+		}
+	}
+
+	// We do not validate tags because they have no explicit rules
+	if err := Validate(dw.Tags); err != nil {
+		retVal = multierror.Append(retVal, err)
+	}
+
+	retVal = validatePercent(retVal, dw.Weight, "weight")
+
+	return retVal
+}
+
+// ValidateHTTPTimeout validates HTTP Timeout
+func ValidateHTTPTimeout(timeout *proxyconfig.HTTPTimeout) error {
+	var retVal error
+
+	if simple := timeout.GetSimpleTimeout(); simple != nil {
+		if simple.TimeoutSeconds < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("timeout_seconds must be in range [0..]"))
+		}
+
+		// We ignore override_header_name
+	}
+
+	return retVal
+}
+
+// ValidateHTTPRetries validates HTTP Retries
+func ValidateHTTPRetries(retry *proxyconfig.HTTPRetry) error {
+	var retVal error
+
+	if simple := retry.GetSimpleRetry(); simple != nil {
+		if simple.Attempts < 0 {
+			retVal = multierror.Append(retVal, fmt.Errorf("attempts must be in range [0..]"))
+		}
+
+		// We ignore override_header_name
+	}
+
+	return retVal
+}
+
+// ValidateHTTPFault validates HTTP Fault
+func ValidateHTTPFault(fault *proxyconfig.HTTPFaultInjection) error {
+	var retVal error
+
+	if fault.GetDelay() != nil {
+		if err := validateDelay(fault.GetDelay()); err != nil {
+			retVal = multierror.Append(retVal, err)
+		}
+	}
+
+	if fault.GetAbort() != nil {
+		if err := validateAbort(fault.GetAbort()); err != nil {
+			retVal = multierror.Append(retVal, err)
+		}
+	}
+
+	return retVal
+}
+
+// ValidateL4Fault validates L4 Fault
+func ValidateL4Fault(fault *proxyconfig.L4FaultInjection) error {
+	var retVal error
+
+	if fault.GetTerminate() != nil {
+		if err := validateTerminate(fault.GetTerminate()); err != nil {
+			retVal = multierror.Append(retVal, err)
+		}
+	}
+
+	if fault.GetThrottle() != nil {
+		if err := validateThrottle(fault.GetThrottle()); err != nil {
+			retVal = multierror.Append(retVal, err)
+		}
+	}
+
+	return retVal
+}
+
+func validateSubnet(subnet string) error {
+	var retVal error
+
+	// TODO verify subnet is "IPv4 or IPv6 ip address with optional subnet. E.g., a.b.c.d/xx form or just a.b.c.d
+
+	return retVal
+}
+
+func validateDelay(delay *proxyconfig.HTTPFaultInjection_Delay) error {
+	var retVal error
+
+	retVal = validateFloatPercent(retVal, delay.Percent, "delay")
+
+	if delay.GetFixedDelaySeconds() < 0 {
```

Comment:
nit: @rshriram we should switch to duration proto. Then you can just say "1s" in JSON.